versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/codecs/L2ToBsonConverter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.mongodb.codecs;
+
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.bson.BsonBinary;
+import org.bson.BsonWriter;
+
+import com.dremio.nessie.versioned.impl.L2;
+import com.dremio.nessie.versioned.store.Entity;
+
+/**
+ * This specifically converts an L2 object into BSON format.
+ */
+public class L2ToBsonConverter {
+
+  private final BsonWriter writer;
+
+  /**
+   * Initializes the writing process to BSON.
+   * @param writer the write into which the object will be serialized.
+   */
+  public L2ToBsonConverter(BsonWriter writer) {
+    this.writer = writer;
+  }
+
+  public void write(L2 l2) {
```

Comment:
You may be thinking this but I'd expect this to be a generic class that simply get's a Schema and Class upon construction. Basically each object codec is just an instance of a generic entity<>bson codec.

Reply:
Done, see my other comment.
https://github.com/projectnessie/nessie/pull/473#discussion_r529879193